fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/lock/MultiUserLock.java

```diff
@@ -75,12 +75,20 @@ private LockGrantType tryLock(LockHolder lockHolderRequest) {
             }
         }
 
-        LockHolder sameLockHolder = null;
         while (lockOwner != null) {
-            if (lockHolderRequest.equals(lockOwner)) {
-                sameLockHolder = lockOwner;
+            boolean isConflict = lockOwner.isConflict(lockHolderRequest);
```

Comment:
what if the same locker plan to acquire a new X lock again after holding a X lock on the same object?